### PR TITLE
Doc: beautify usetex demo example

### PR DESCRIPTION
**Pull request by ImportanceOfBeingErnest**
## PR Summary

Beautify usetex demo. This demo was too crowded and contained unnecessary commands. This PR simplifies and beautifies it in the hope to lower the risk of eye cancer. 

In total I'm not convinced this demo is entirely necessary, but the way I changed it now, it at least shows some interesting things like the combination of math mode and text mode in one text, or the use of `eqnarray` environements. 

**old**

image

**new**
image

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
This is better - any chance the \gamma and \omega could be separated vertically more?

**Comment by ImportanceOfBeingErnest** [comment]
Sure. Updated.

File: examples/text_labels_and_annotations/usetex_demo.py [hunk 1]
```diff
@@ -14,66 +18,55 @@
...
-         'k', linewidth=2)
-plt.plot((-delta / 2, -delta / 2 + offset * 2), (height, height + offset),
-         'k', linewidth=2)
-plt.plot((delta / 2, delta / 2 - offset * 2), (height, height - offset),
-         'k', linewidth=2)
-plt.plot((delta / 2, delta / 2 - offset * 2), (height, height + offset),
-         'k', linewidth=2)
-plt.text(-0.06, height - 0.06, r'$\delta$', {'color': 'k', 'fontsize': 24})
+plt.annotate("",
+            xy=(-delta / 2., 0.1), xycoords='data',
```
Comment (timhoffm) [inline examples/text_labels_and_annotations/usetex_demo.py]: This and the following lines could use one more space.

File: examples/text_labels_and_annotations/usetex_demo.py [hunk 2]
```diff
@@ -3,7 +3,11 @@
 Usetex Demo
 ===========
 
+Shows how to use latex in a plot. 
+
+Also refer to the :doc:`/tutorials/text/usetex` guide.
 """
+
 import matplotlib
 matplotlib.rc('text', usetex=True)
```
Comment (timhoffm) [inline examples/text_labels_and_annotations/usetex_demo.py]: Not your fault, but I would do the rc change after all the imports.

**Comment by ImportanceOfBeingErnest** [comment]
Updated and passed the tests.

**Comment by QuLogic** [comment]
This example doesn't pass PEP8 (see backport's [build failure](https://travis-ci.org/matplotlib/matplotlib/jobs/394783279)), but I don't know why master build is not seeing it.

**Comment by QuLogic** [comment]
The PEP8 checks should be re-enabled on master by #11477.

**Pull request merged**